exonum-java-binding/qa-service/src/main/java/com/exonum/binding/qaservice/QaServiceImpl.java

```diff
@@ -231,4 +217,37 @@ private HashCode submitTransaction(RawTransaction rawTransaction) {
   private void checkBlockchainInitialized() {
     checkState(node != null, "Service has not been fully initialized yet");
   }
+
+  @Override
+  public void verifyConfiguration(Fork fork, Configuration configuration) {
+    InitialConfiguration config = configuration.getAsMessage(InitialConfiguration.class);
+    checkConfiguration(config);
+  }
+
+  @Override
+  public void applyConfiguration(Fork fork, Configuration configuration) {
+    updateTimeOracle(fork, configuration);
+  }
+
+  private void checkConfiguration(InitialConfiguration config) {
```

Comment:
Shouldn't we rename `InitialConfiguration` if it's not necessarily `initial`?

Reply:
Good observation, thanks, let it be QaConfiguration.

(@ivan-ochc — see the patches to the specification: https://wiki.bf.local/pages/viewpreviousversions.action?pageId=1710016)